src/os/bluestore/BlueStore.cc

```diff
@@ -2037,10 +2073,14 @@ int BlueStore::fsck()
   if (r < 0)
     goto out_bdev;
 
-  r = _open_alloc();
+  r = _open_fm(false);
   if (r < 0)
     goto out_db;
 
+  r = _open_alloc();
+  if (r < 0)
+    goto out_fm;
+
   r = _open_super_meta();
```

Comment:
We shall move ```_open_super_meta()``` ahead here,  just before calling ```_open_fm(false)```(as we did in ```mount()```), because ```_open_fm(false)``` relies on **freelist_type** to be preloaded from db.